pkg/cloudprovider/fake/capacity.go

```diff
@@ -54,8 +54,8 @@ func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.Constr
 	}}, nil
 }
 
-func (c *Capacity) Delete(ctx context.Context, nodes []*v1.Node) error {
-	return nil
+func (c *Capacity) Delete(ctx context.Context, nodes []*v1.Node) (map[string]bool, error) {
```

Comment:
I think we should avoid returning this `map` - I think it's too complicated to expect the calling code to really know what to do about the problem. Either it's a persistent problem and a human operator needs to get involved (some node is "stuck" being terminated), or else it's a transient problem and will go away on its own in the next few rounds of reconciliation. So really all the code calling `Delete` can do is log - but the cloud provider code could just as easily log as well (and then you don't need `GetInstanceId`.